### DISCUSSION TITLE
EntityPointer-like Resource

### DISCUSSION THREAD

**Discussion by DasLixou** [bevyengine/bevy#6514] [Q&A] [answered]
Hi! I wonder if there is a simpler way of making a Resource marked struct containing an entity and then querying normally and then using the id from the resource in that query. or short: is there a thing similar to resource which can be applied to only one entity at a time and can be directly put on a query so the scheduler also can handle it that way (for async things etc)?

**Comment by DasLixou** [comment, marked as answer]
EDIT: this would be very cool, because currently i get an error that the queries couldn't be merged into a ParamSet because `Query<&Transform>` and `Query<&mut Transform, With<Camera>>` yeah... there is no HasPointer<CameraFollow> or smth :)

**Reply by tim-blackbird** [reply]
If you do `Query<&Transform, Without<Camera>>` for the first query it shouldn't error anymore :)

**Reply by DasLixou** [reply]
But is there some kind of EntityPointerResourceMarker-like thing?

**Reply by rparrett** [reply]
You can store [`Entity`](https://docs.rs/bevy/latest/bevy/ecs/entity/struct.Entity.html) ("entity ids") in a resource (or a component) and use [`Query::Get`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#method.get) and related methods.

```rust
#[derive(Default, Deref, DerefMut)]
struct Target(Option<Entity>);

fn startup_setup(mut target: ResMut<Target>, mut commands: Commands) {
    let entity = commands.spawn().id();
    **target = Some(entity);
}

fn system(target: Res<Target>, query: Query<&Transform>) {
    if let Some(transform) = target.and_then(|e| query.get(e).ok()) {
        // ...
    }
}
```

**Reply by DasLixou** [reply]
Just one more question: when I have a system with Query<&Transform> where I do the stuff with the entity and for example another system also using Query<&Transform, With<Player>> then they can’t run in parallel, correct? But when there would be something like a marker resource component where we do Query<&Transform, Marked<Stuff>> then they could run parallel couldn’t they?

**Reply by tim-blackbird** [reply]
> Two systems cannot be executed in parallel if both access a certain component and at least one of the accesses is mutable, unless the schedule can verify that no entity could be found in both queries, as otherwise Rusts mutability Rules would be broken.

From [here](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#mutable-component-access).

**Reply by rparrett** [reply]
I believe that's roughly correct (although `&Transform` and `&Transform` are read-only access, so those can be parallelized. I'm assuming you've just omitted a `mut` there.)

But you may need to be more explicit like
`Query<(&mut Transform, Marker)>` + `Query<&Transform, Without<Marker>>` to express that sort of thing.

This abandoned-for-now draft chapter for the new bevy book explains this in extreme detail: https://github.com/bevyengine/bevy-website/blob/d5676eee0b06e579328553f617cc8847ab589ebf/content/learn/book/game-logic/system-ordering/_index.md#advanced-data-access-and-archetypes

This was [summarized by cart](https://github.com/bevyengine/bevy/discussions/3119#discussioncomment-1637393) like
> If two Queries both access the same component and one or both of them mutates the component, they will conflict. This conflict can be avoided if one of the Queries adds a Without filter for one or more of the components in the other Query.

**Reply by DasLixou** [reply]
Oh ok. But when you said
> unless the schedule can verify that no entity could be found in both queries

That means that i can’t really say in the systems signature that I just want the entity from the resource? The only possibility is to use Without<> but when I don’t know where I’ll use it and where i won’t that can be very tricky I would say
